### ISSUE TITLE
Training Signal Annealing Implementation

### ISSUE THREAD

**Issue by nazim1021** [peisuke/UnsupervisedDataAugmentation.pytorch#1]
Hi, 
First all thanks and great work getting started on this paper. I wanted to ask if you have been able to work on supervised loss as per paper?  I have implemented a small function on train signal annealing along with new supervised loss, would be great if you can have a look. Additionally, I updated the code to use WideResNet 28-2 as mentioned in paper. Although I am not getting good results yet. Would happy to collaborate with you on further work :)

**Comment by peisuke** [comment]
Thanks for your comment. First of all, I have only little progress. Recently, I have enjoyed VR and embedding system, so I did not pay time for ML. Next week, I will make some progress in this work!
If you have some progress for this paper's work, I welcome to your advice.

**Comment by nazim1021** [comment]
@peisuke oh wow VR sounds amazing.  So, with regards to this paper, I have done some work. I have updated the code to include TSA loss along with options to use tsa schedule. Also updated model to WideResnet 28-2. After running for 10000 iterations, the training accuracy is pretty great, although the test accuracy seems to be always around 68%. Not sure if i have missed something from the paper. Below is my implementation

![image](https://user-images.githubusercontent.com/39544613/59566958-d2ffb180-901b-11e9-8c4e-d80d29e4832b.png)

Let me know. Would happy to contribute more on this

**Comment by getsanjeev** [comment]
@nazim1021 Can you please look into https://github.com/peisuke/UnsupervisedDataAugmentation.pytorch/issues/2

**Comment by preachermaan** [comment]
Thanks guys. Just noticed that model.train() is outside the training loop so will be switched off after the first eval.
